sdk/python/feast/diff/FcoDiff.py

```diff
@@ -0,0 +1,109 @@
+from dataclasses import dataclass
+from enum import Enum
+from typing import Any, List, Set, Tuple, Union
+
+from feast.entity import Entity
+from feast.feature_service import FeatureService
+from feast.feature_table import FeatureTable
+from feast.feature_view import FeatureView
+from feast.on_demand_feature_view import OnDemandFeatureView
+from feast.request_feature_view import RequestFeatureView
+
+
+@dataclass
+class PropertyDiff:
+    property_name: str
+    val_existing: str
+    val_declared: str
+
+
+class TransitionType(Enum):
+    UNKNOWN = 0
+    CREATE = 1
+    DELETE = 2
+    UPDATE = 3
+    UNCHANGED = 4
+
+
+@dataclass
+class FcoDiff:
+    current_fco: Any
+    new_fco: Any
+    fco_property_diffs: List[PropertyDiff]
+    transition_type: TransitionType
+
+
+class RegistryDiff:
+    fco_diffs: List[FcoDiff]
+
+    def __init__(self):
+        self.fco_diffs = []
+
+    def add_fco_diff(self, fco_diff: FcoDiff):
+        self.fco_diffs.append(fco_diff)
+
+
+def _tag_registry_entities_for_keep_delete(
+    existing_entities: Set[Entity], desired_entities: Set[Entity]
+) -> Tuple[Set[Entity], Set[Entity], Set[Entity]]:
```

Comment:
in fact, can all of these methods be combined into one? e.g. `_tag_registry_objects_for_keep_delete`? seems like the internal logic is the exact same

Reply:
Yeah it's the same. I tried to do it but it made pypy very unhappy, but I can clean it up a bit and rename.